### PR TITLE
Create service to restart wazuh

### PR DESCRIPTION
**Pull request by yenienserrano**
**Description:**

The way to restart the manager/cluster was centralized in a service, since each view that had the option to restart the manager/cluster did it in a different way, 
The delay of the `/cluster/restart` request was removed and the delay was added before the request to wait for all the nodes to synchronize.
Removed code that is no longer used

**Issue:**

- #4277 

**Test:**

To test it you have to go to any page that has the restart option and try doing the restart, when it tells you that it has restarted try navigating between pages that make requests to the API.

_Successful restart_

1. Navigate to Management>Configuration>Edit configuration || Management>Status || Management>Rules||Decoders||CDB lists and edit file 
2. Click on "Restart cluster||Restart Manager" and wait for the modal to disappear
3. Navigate to another route that uses some request for the API

_Restart no complite_

1. Navigate to Management>Configuration>Edit configuration || Management>Status || Management>Rules||Decoders||CDB lists and edit file 
2. Click on "Restart cluster||Restart Manager".
3. When you make the first attempt to remove the container from the manager
4. Wait for all attempts, change the modal and redirect to healthcheck

**Screenshot:**

Updated screenshots in comments

https://user-images.githubusercontent.com/63758389/181754945-d74a5310-6a4a-409a-881b-0365958e1e68.mp4

### CODE REVIEW

**Comment by yenienserrano** [comment]
Added a modal so that the user cannot navigate during Wazuh restart

https://user-images.githubusercontent.com/63758389/182565501-f833d889-7d29-483e-a95e-578d3124fb53.mp4

**Comment by Mayons95** [comment]
TR :white_check_mark: 
[Screencast from 03-08-22 07:46:05.webm](https://user-images.githubusercontent.com/104914131/182590942-7cb33e04-8868-40ab-a13d-470ba685efba.webm)
[Screencast from 03-08-22 07:39:17.webm](https://user-images.githubusercontent.com/104914131/182590949-44398b19-efec-48cb-be9a-cf6477182e9e.webm)

C.C @yenienserrano

**Review by Mayons95 — APPROVED** [review]
Test OK

**Comment by yenienserrano** [comment]
Modal sync:
Sync

Modal error
error

Modal attempt
attempt

Restart:

https://user-images.githubusercontent.com/63758389/183454919-8578a45e-7b4f-4485-804f-55f375d096ef.mp4

Not restart:

https://user-images.githubusercontent.com/63758389/183454925-f4ca638a-2dd6-42da-947c-bdd02556f80f.mp4

File: public/components/common/restart-modal/restart-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  // Component props
+  const { isRestarting: isRestartingProp, useDelay = false } = props;
+
+  // Modal's restart state
+  const [currentState, setCurrentState] = useState(RESTART_STATES.RESTARTING);
+
+  // Restart attempt
+  const [restartAttempt, setRestartAttempt] = useState(
+    store.getState().appStateReducers.restartAttempt
+  );
```
Comment (Desvelao) [inline public/components/common/restart-modal/restart-modal.tsx:41]: **nitpick**: Why are we using a global state to set the component state?

We should not use the Redux store directly. If we need a global state store in Redux, then we use the `useSelector` hook.

File: public/controllers/management/components/management/status/actions-buttons-main.js [hunk 2]
```diff
@@ -48,68 +50,37 @@ class WzStatusActionButtons extends Component {
...
-        'Restarting cluster, it will take up to 30 seconds.',
-        3000
-      );
+      await RestartHandler.restartWazuh(this.props.updateRestartAttempt);
+      this.setState({ isRestarting: false, timeoutRestarting: false });
     } catch (error) {
       this.setState({ isRestarting: false });
       const options = {
-        context: `${WzStatusActionButtons.name}.restartCluster`,
+        context: isCluster ? `${WzStatusActionButtons.name}.restartCluster` : `${WzStatusActionButtons.name}.restartManager`,
```
Comment (Desvelao) [inline public/controllers/management/components/management/status/actions-buttons-main.js:77]: **nitpick**: `restartManager` in the log doesn't match with the method name `restartWazuh`.

**Comment by yenienserrano** [comment]
**Test cases:**

Restart successful:
- [x] Restart correctly
- [x] Synchronizes cluster and restarts correctly
- [x] Restart node successfully

Restart error:
- [x] Failed to restart
- [x] Failed to synchronize after successful restart
- [x] Synchronize failed after error on restart
- [x] Synchronize successful and error on restart
- [x] Error restarting node

**Restart correctly:**

https://user-images.githubusercontent.com/63758389/184157036-b3125a4d-c994-4b45-a8dc-969dd472f277.mp4

**Synchronizes cluster and restarts correctly:**

https://user-images.githubusercontent.com/63758389/184157195-d753d559-7844-49e4-a67d-3adf2d732841.mp4

**Restart node successfully:**

https://user-images.githubusercontent.com/63758389/184157403-25ba89f0-3d53-4587-b05d-ad7d2d69dbe2.mp4

**Failed to restart:**

https://user-images.githubusercontent.com/63758389/184157634-20e3b210-97bf-49a3-b67e-1168f94bb55f.mp4

**Failed to synchronize after successful restart:**

https://user-images.githubusercontent.com/63758389/184157909-3e227807-9b08-44c7-8b14-c9e3e53b57f4.mp4

**Synchronize failed after error on restart:**

https://user-images.githubusercontent.com/63758389/184158330-6d939402-d847-46b7-905e-61614ad1e93e.mp4

**Synchronize successful and error on restart:**

https://user-images.githubusercontent.com/63758389/184158492-d490af49-bb26-43a1-9ed6-28732ecb3b94.mp4

**Error restarting node:**

https://user-images.githubusercontent.com/63758389/184158594-c70125e1-3ca6-4305-9c80-304da4490e90.mp4

File: public/components/common/restart-modal/restart-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        ),
+        actions: (
+          <EuiFlexGroup justifyContent="flexEnd">
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty color='danger' flush="both" onClick={forceRestart}>
+                Force restart
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}> 
+              <EuiButton color="primary" fill onClick={() => dispatch(updateRestartStatus(RestartHandler.RESTART_STATES.RESTARTED))}>
```
Comment (AlexRuiz7) [inline public/components/common/restart-modal/restart-modal.tsx:165]: Move this code to a function named `abort` that does all the cleansing (sets everything to the proper state).
```suggestion
              <EuiButton color="primary" fill onClick={abort}>
```

**Review by asteriscos — APPROVED** [review]
CR: :heavy_check_mark: 
Test: :heavy_check_mark: 

Peek 2022-08-12 11-36

**Comment by asteriscos** [comment]
Worker restarting error are properly handled :heavy_check_mark: 

image

**Review by chantal-kelm — APPROVED** [review]

**Pull request merged**